library/std/src/sys/pal/windows/mod.rs

```diff
@@ -289,6 +288,14 @@ pub fn truncate_utf16_at_nul(v: &[u16]) -> &[u16] {
     }
 }
 
+pub fn ensure_no_nuls<T: AsRef<OsStr>>(s: T) -> crate::io::Result<T> {
+    if s.as_ref().encode_wide().any(|b| b == 0) {
```

Comment:
Hm, are we using this for soundness anywhere? The `AsRef<OsStr>` conversion could e.g. return different results spuriously which would be a problem if we are.

Reply:
Luckily that's not the case, we only use this with trusted types. But I agree that this isn't great, I'll try to find a better solution in another PR.